web_src/js/features/clipboard.js

```diff
@@ -16,6 +16,36 @@ function onError(btn) {
   btn.dataset.content = btn.dataset.original;
 }
 
+/** Use the document.execCommand to copy the value to clipboard */
+function fallbackCopyViaSelect(elem) {
+  elem.select();
+
+  // if unsecure (not https), there is no navigator.clipboard, but we can still use document.execCommand to copy to clipboard
+  // it's also fine if we don't test it exists because of the try statement
+  return document.execCommand('copy');
+}
+/**
+ * Fallback to use if navigator.clipboard doesn't exist.
+ * Achieved via creating a temporary textarea element, selecting the text, and using document.execCommand.
+ */
+function fallbackCopyToClipboard(text) {
+  const tempTextArea = document.createElement('textarea');
+  tempTextArea.value = text;
+
+  // avoid scrolling
+  tempTextArea.style.top = 0;
+  tempTextArea.style.left = 0;
+  tempTextArea.style.position = 'fixed';
+
+  document.body.appendChild(tempTextArea);
+
+  const success = fallbackCopyViaSelect(tempTextArea);
+
+  document.body.removeChild(tempTextArea);
+
+  return success;
+}
```

Comment:
I don't think this fallback is necessary for the browsers we support. I think all of them can do `document.execCommand`.

Reply:
Thought I'd argue that we sometimes will want to copy stuff the the clipboard that is not contained in a `<input>`, so I think it'd be better to only use this method with a temporary `<textarea>` to copy arbitrary text and remove the `document.execCommand` variant (it is deprecated API anyways).

Reply:
> Thought I'd argue that we sometimes will want to copy stuff the the clipboard that is not contained in a `<input>`, so I think it'd be better to only use this method with a temporary `<textarea>` to copy arbitrary text and remove the `document.execCommand` variant (it is deprecated API and the fact that it messes with UX by changing the selection is not good either).

Yeah, I just thought that using the existing `<input>` elements was better than making the temporary `<textarea>`, but I'll change it to only use the temporary `<textarea>` method

Reply:
That's fine. I just want to avoid visible UX interactions like `inputElement.select()`, so using the temporary textarea is preferred.